content/toolsAndResources/nge/ngeBlocks.md

```diff
@@ -734,15 +1411,119 @@ This function converts the input value from degrees to radians. This node can su
+- **Execution count** shows the number of times this node is executed. This value will affect the performance of the graph. Depending on the value of *Evaluate context* on nodes further in the graph the values for *Execution count* may be lower than the value for *Call count*. 
+
+#### Inputs
+- **left** requires a Vector3 or Vector4 to represent the first vector.
+- **right** requires a Vector3 or Vector4 to represent the second vector.
+
+#### Outputs
+- **output** is a Vector3 that represents a vector that is perpendicular to both the left and right vectors. 
+
+<H3Image title="Curve" image="/img/tools/nge/curve.jpg" alt="Curve node"/>
+This node will apply a specific chosen curve function to the input data to modify it based on the applied curve. This node supports Float, Int, Vector2, Vector3, and Vector4 data types.
+
+#### Debug Infos
+- **Call count** shows the number of times this node is called by nodes further into the graph. This is the total number of calls from all subsequent nodes. 
+- **Execution count** shows the number of times this node is executed. This value will affect the performance of the graph. Depending on the value of *Evaluate context* on nodes further in the graph the values for *Execution count* may be lower than the value for *Call count*. 
+
+#### Advanced
+- **Type** defines the curve function to apply to the input data from a pre-polulated list of functions. 
+
+#### Inputs
+- **input** is connected to a node providing the input data for the function.
+
+#### Outputs
+- **output** is the transformation of the input data by a specific curve function and will match the type of the input data. If the input data has multiple components, the curve function will be applied to each component individually. 
+
+<H3Image title="Distance" image="/img/tools/nge/distance.jpg" alt="Distance node"/>
+This node will calculate the distance between two coordinates. This node supports Vector2, Vector3, and Vector4 data types.
+
+#### Debug Infos
+- **Call count** shows the number of times this node is called by nodes further into the graph. This is the total number of calls from all subsequent nodes. 
+- **Execution count** shows the number of times this node is executed. This value will affect the performance of the graph. Depending on the value of *Evaluate context* on nodes further in the graph the values for *Execution count* may be lower than the value for *Call count*. 
+
+#### Inputs
+- **left** provides the starting coordinate for the function.
+- **right** provides the ending coordinate for the function.
+
+#### Outputs
+- **output** is the distance between the left and right inputs represented as a value of Float type.
+
+<H3Image title="Dot" image="/img/tools/nge/dot.jpg" alt="Dot node"/>
+This node will calculate the cosine of the angle between two Euclidean vectors. This node supports Vector2, Vector3, and Vector4 data types.

```

Comment:
In fact, it will be the cosine only if the two vectors are normalized. I think you should say that it calculates the dot product of the two inputs, and that it is the cosine of the angle between the vectors when the vectors are both normalized.

Reply:
Updated